src/linear_algebra/matrix/circulant.lean

```diff
@@ -0,0 +1,255 @@
+/-
+Copyright (c) 2021 Lu-Ming Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lu-Ming Zhang
+-/
+import linear_algebra.matrix.symmetric
+import data.polynomial.monomial
+import data.matrix.pequiv
+import data.equiv.fin
+
+/-!
+# Circulant matrices
+
+This file contains the definition and basic results about circulant matrices.
```

Comment:
Can you please include a definition of circulant matrix in this first section of the module docstring?